Library/Formula/node.rb

```diff
@@ -14,17 +14,15 @@ class Node < Formula
   option "with-debug", "Build with debugger hooks"
   option "without-npm", "npm will not be installed"
   option "without-completion", "npm bash completion will not be installed"
+  option "with-full-icu", "Build with full-icu (all locales) instead of small-icu (English only)"
 
   deprecated_option "enable-debug" => "with-debug"
+  deprecated_option "with-icu4c" => "with-full-icu"
```

Comment:
Not sure if we should keep this or not:
* argument for keeping this: no side effect for someone who has previously build node `--with-icu4c` and is relying on non English locales (through he could add other locales at runtime to a `small-icu` build as mentioned above)
* argument against keeping this: someone who needs only the English local and has previously build node `--with-icu4c` would have an unnecessary big node installation and would have to build it from source instead of downloading a smaller bottle